examples/robot_bodybrain_ea/plot_fitness.py

```diff
@@ -0,0 +1,45 @@
+"""Plot the fitness over generations for a previous optimization process."""
+
+import logging
+
+import config
+import matplotlib.pyplot as plt
+import pandas
+from generation import Generation
+from individual import Individual
+from population import Population
+from revolve2.core.database import OpenMethod, open_database_sqlite
+from sqlalchemy import select
+
+
+def main() -> None:
+    """Run the program."""
+    logging.basicConfig(
+        level=logging.INFO,
+        format="[%(asctime)s] [%(levelname)s] [%(module)s] %(message)s",
+    )
+
+    db = open_database_sqlite(config.DATABASE_FILE, OpenMethod.OPEN_IF_EXISTS)
+    df = pandas.read_sql(
+        select(Generation.generation_index, Individual.fitness)
+        .join(Generation.population)  # type: ignore[misc] # TODO must be possible to type this properly
+        .join(Population.individuals),  # type: ignore[misc] # TODO must be possible to type this properly
+        db,
+    )
+
+    agg = df.groupby("generation_index")["fitness"].agg(["mean", "min", "max"])
+    plt.figure()
+    plt.plot(agg.index, agg["mean"], label="Average Fitness")
+    plt.plot(agg.index, agg["min"], label="Min Fitness")
+    plt.plot(agg.index, agg["max"], label="Max Fitness")
+
+    plt.xlabel("Generation Index")
+    plt.ylabel("Fitness")
+    plt.title("Fitness Over Generations")
+    plt.legend()
+
+    plt.show()
```

Comment:
I got the below warning when running this line

> UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.

I don't know which GUI backend you use? `tkinter`?

Reply:
I did not set anything. I am not sure what the default is(maybe depends on the running system?)

Reply:
indeed i just tried and removing it breaks the example

Reply:
So I looked and tk simply needs to be installed through the users package manager or else it comes with the python installation sometimes. It doesn't appear to be something that can be installed with pip. So I guess we will have to add that to a tutorial at some point but for now we can't do much in the package.